cmd/dfget/app/server.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The Dragonfly Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/dragonflyoss/Dragonfly/dfget/core/uploader"
+	"github.com/dragonflyoss/Dragonfly/dfget/util"
+	"github.com/spf13/cobra"
+)
+
+var serverCmd = &cobra.Command{
+	Use:   "server",
+	Short: "Launch a peer server for uploading files.",
+	Run: func(cmd *cobra.Command, args []string) {
+		logPath := filepath.Join(cfg.WorkHome, "logs")
+		cfg.ServerLogger = util.CreateLogger(logPath,
+			"dfserver.log", "INFO", cfg.Sign)
+		if port, err := uploader.LaunchPeerServer(cfg); err == nil {
+			fmt.Println(port)
+			uploader.WaitForShutdown()
+		} else {
+			fmt.Println(err.Error())
+		}
+	},
+}
+
+func init() {
+	initServerFlags()
+	rootCmd.AddCommand(serverCmd)
+}
+
+func initServerFlags() {
+	serverCmd.PersistentFlags().StringVar(&cfg.RV.SystemDataDir, "data", cfg.RV.SystemDataDir,
+		"the directory which stores temporary files for p2p uploading")
+	serverCmd.PersistentFlags().StringVar(&cfg.WorkHome, "home", cfg.WorkHome,
+		"the work home of dfget server")
+	serverCmd.PersistentFlags().StringVar(&cfg.RV.LocalIP, "ip", "",
+		"the ip that server will listen on")
+	serverCmd.PersistentFlags().IntVar(&cfg.RV.PeerPort, "port", 0,
```

Comment:
Is it reasonable to config the default port to be 0?
How about a fixed port number?

Reply:
`0` means that `dfget server` will auto-generate a port and attempt to listen and serve on this port. It will retry at most 10 times if `dfget server` starts failed.